src/hotspot/share/gc/shenandoah/shenandoahScanRemembered.inline.hpp

```diff
@@ -772,7 +772,10 @@ void ShenandoahScanRemembered<RememberedSet>::process_clusters(size_t first_clus
     } else {
       // ==== BEGIN CLEAN card range processing ====
 
-      assert(ctbm[cur_index] == CardTable::clean_card_val(), "Error");
+      // If we are using the write table (during update refs, e.g.), a mutator may dirty
+      // a card at any time. This is fine for the algorithm below because it is only
```

Comment:
Didn't finish your thought in this comment.

Reply:
Hmm, strange. Fixed it.

Reply:
The weakening of the assert and the explanatory comment look good to me; thanks!